From 7d3edb20b400abf08f3f2465b6662074662f0529 Mon Sep 17 00:00:00 2001
From: Jordan Hendricks <jordan.hendricks@joyent.com>
Date: Mon, 5 Mar 2018 23:15:45 +0000
Subject: [PATCH] MANTA-3597 `mpicker choose` does not convert input size to
 bytes Reviewed by: Kelly McLaughlin <kelly.mclaughlin@joyent.com> Reviewed
 by: Tim Kordas <tim.kordas@joyent.com> Approved by: Tim Kordas
 <tim.kordas@joyent.com>

---
 bin/mpicker | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/bin/mpicker b/bin/mpicker
index 9affeff..d7dca66 100755
--- a/bin/mpicker
+++ b/bin/mpicker
@@ -356,7 +356,8 @@ MPicker.prototype.do_poll.help = [
  */
 MPicker.prototype.do_choose = function do_choose(subcmd, opts, args, cb) {
     var self = this;
-    var size = opts.size;
+    var sizeMB = opts.size;
+    var sizeBytes = sizeMB * 1048576;
     var replicas = opts.replicas;
     var dcs = [];
 
@@ -376,7 +377,7 @@ MPicker.prototype.do_choose = function do_choose(subcmd, opts, args, cb) {
         var summary = [];
 
         summary.push('replicas requested: ' + replicas);
-        summary.push('size requested: ' + size + ' MB');
+        summary.push('size requested: ' + sizeMB + ' MB');
 
         var allSharkCount = 0;
         var inUseSharkCount = 0;
@@ -506,8 +507,9 @@ MPicker.prototype.do_choose = function do_choose(subcmd, opts, args, cb) {
             p.setRecordsDb(user_db, function () {
                 var chooseOpts = {
                     replicas: replicas,
-                    size: size
+                    size: sizeBytes
                 };
+
                 p.choose(chooseOpts, function (err, sharks, chooseStats) {
                     if (err) {
                         if (err.cause().name === 'PickerError') {
-- 
2.21.0

